Components/Pages/Goals.js

```diff
@@ -30,14 +35,77 @@ const Goals = ({ navigation }) => {
   const { goals, refetch } = useGoalsPull();
   const { updateGoal } = useGoalUpdate();
   const { removeGoal } = useGoalDelete();
+  const [notification, setNotification] = useState({});
+  const [expoPushToken, setExpoPushToken] = useState('');
+
+  const registerForPushNotificationsAsync = async () => {
+    if (Constants.isDevice) {
+      const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
+      let finalStatus = existingStatus;
+      if (existingStatus !== 'granted') {
+        const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
+        finalStatus = status;
+      }
+      if (finalStatus !== 'granted') {
+        alert('Failed to get push token for push notification!');
+        return;
+      }
+      let token = await Notifications.getExpoPushTokenAsync();
+      console.log(token);
+      setExpoPushToken(token);
+    } else {
+      alert('Must use physical device for Push Notifications');
+    }
+
+    if (Platform.OS === 'android') {
+      Notifications.createChannelAndroidAsync('default', {
+        name: 'default',
+        sound: true,
+        priority: 'max',
+        vibrate: [0, 250, 250, 250],
+      });
+    }
+  };
+
+  const _handleNotification = notification => {
+    Vibration.vibrate();
+    console.log(notification);
+    this.setState({ notification: notification });
```

Comment:
It looks like you should be using setNotification - since you defined that above. Also I think you can only use this.setState in class-based components

Reply:
Good catch!